nifi-nar-bundles/nifi-alluxio-bundle/nifi-alluxio-processors/src/main/java/org/apache/nifi/processors/alluxio/AbstractAlluxioProcessor.java

```diff
@@ -0,0 +1,113 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.alluxio;
+
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * AbstractCassandraProcessor is a base class for Alluxio processors and contains logic and variables common to most
+ * processors integrating with Alluxio.
+ */
+public abstract class AbstractAlluxioProcessor extends AbstractProcessor {
+
+    public static final PropertyDescriptor MASTER_HOSTNAME = new PropertyDescriptor.Builder()
+            .name("alluxio-master-ip")
+            .displayName("Master hostname")
+            .description("Hostname of the Alluxio File System Master node.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MASTER_PORT = new PropertyDescriptor.Builder()
+            .name("alluxio-master-port")
+            .displayName("Master port")
+            .description("Port to use when connecting to the Alluxio File System Master node.")
+            .required(true)
+            .defaultValue("19998")
+            .addValidator(StandardValidators.PORT_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor URI = new PropertyDescriptor.Builder()
+            .name("alluxio-uri")
+            .displayName("URI")
+            .description("Alluxio URI to use. Example: /path")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+    static {
+        descriptors.add(MASTER_HOSTNAME);
+        descriptors.add(MASTER_PORT);
+        descriptors.add(URI);
+    }
+
+    protected final AtomicReference<FileSystem> fileSystem = new AtomicReference<>(null);
+
+    protected void createFileSystem(ProcessContext context) {
+        System.setProperty("alluxio.master.hostname", context.getProperty(MASTER_HOSTNAME).getValue());
+        System.setProperty("alluxio.master.port", context.getProperty(MASTER_PORT).getValue());
+        fileSystem.set(FileSystem.Factory.get());
```

Comment:
This doesn't seem to update regardless of changing a property.  I accidentally used the wrong port when testing and despite changing in the properties, stopping/starting or disabling/enabling, the backing Alluxio client seems stuck on the initial port.